biocrnpyler/mixture.py

```diff
@@ -303,63 +303,55 @@ def get_parameter(self, mechanism, part_id, param_name):
+            mechanisms = "initial concentration"
+            part_id = mixture.name
+            parameter_name = str(s)
+
+            if s == component.get_species, also checks with parameter_name=component.name
+
+        3. Defaults to 0
         """
+        if isinstance(S, Species):
+            S = [S]
 
-        if not isinstance(s, Species):
-            raise ValueError(f"{s} is not a Species! Can only set initial concentration of a Species.")
-
-        init_conc = None
-        #1
-        if component is not None:
-            init_conc = component.get_initial_condition(s)
-
-        if init_conc is None:
-            #2
-            if (self.name, repr(s)) in self.initial_condition_dictionary:
-                init_conc = self.initial_condition_dictionary[(self.name, repr(s))]
-            #3
-            elif repr(s) in self.initial_condition_dictionary:
-                init_conc = self.initial_condition_dictionary[repr(s)]
-            #4
-            elif component is not None and component.get_species() == s and (self.name, component.name) in self.initial_condition_dictionary:
-                return self.initial_condition_dictionary[(self.name, component.name)]
-            #5
-            elif component is not None and component.get_species() == s and component.name in self.initial_condition_dictionary:
-                return self.initial_condition_dictionary[component.name]
-            #6
-            elif self.parameter_database.find_parameter(None, self.name, repr(s)) is not None:
-                init_conc = self.parameter_database.find_parameter(None, self.name, repr(s)).value
-            #7
-            elif self.parameter_database.find_parameter(None, None, repr(s)) is not None:
-                init_conc = self.parameter_database.find_parameter(None, None, repr(s)).value
-            #8
-            elif component is not None and component.get_species() == s and (None, self.name, component.name) in self.parameter_database:
-                return self.parameter_database.find_parameter(None, self.name, component.name).value
-            #9
-            elif component is not None and component.get_species() == s and component.name in self.parameter_database:
-                return self.parameter_database.find_parameter(None, None, component.name).value
-            #10
-            else:
-                init_conc = 0
+        init_cond_dict = {}
+        for s in S:
+            if not isinstance(s, Species):
+                raise ValueError(f"{s} is not a Species! Can only find initial concentration of a Species.")
+
+            init_cond = None
+            #1 Check the component
+            if component is not None:
+                init_cond = component.get_parameter(param_name = str(s), part_id = self.name, mechanism = "initial concentration", check_mixture = False, return_none = True)
 
-        s.initial_concentration = init_conc
+                if init_cond is None and component.get_species() == s:
```

Comment:
it's called init_cond here. Is it intentional?

Reply:
I called it initial_condition of a CRN and initial concentration inside the ParameterDatabase. I tried to be consistent in that regard to keep back compatibility with Component.initial_concentration.

In my head, CRNs have an initial condition because they can be deterministic or stochastic. However, I figure that calling parameters initial concentration will be more helpful to users.

Reply:
I think having a consistent naming convention in the code helps with code maintenance. Even if certain names are not scientifically accurate.

Reply:
Ok, I will change to initial_concentration everywhere